pkg/apis/config/store.go

```diff
@@ -50,19 +53,43 @@ func FromContextOrDefaults(ctx context.Context) *Config {
 	if cfg := FromContext(ctx); cfg != nil {
 		return cfg
 	}
-	defaults, _ := NewDefaultsFromMap(map[string]string{})
-	featureFlags, _ := NewFeatureFlagsFromMap(map[string]string{})
-	artifactBucket, _ := NewArtifactBucketFromMap(map[string]string{})
-	artifactPVC, _ := NewArtifactPVCFromMap(map[string]string{})
-	metrics, _ := newMetricsFromMap(map[string]string{})
-	trustedresources, _ := NewTrustedResourcesConfigFromMap(map[string]string{})
+	defaults, errDefaults := NewDefaultsFromMap(map[string]string{})
+	if errDefaults != nil {
+		logging.FromContext(ctx).Info(errDefaults)
+	}
+	featureFlags, errFeatureFlags := NewFeatureFlagsFromMap(map[string]string{})
+	if errFeatureFlags != nil {
+		logging.FromContext(ctx).Info(errFeatureFlags)
+	}
+	artifactBucket, errArtifactBucket := NewArtifactBucketFromMap(map[string]string{})
+	if errArtifactBucket != nil {
+		logging.FromContext(ctx).Info(errArtifactBucket)
+	}
+	artifactPVC, errArifactPVC := NewArtifactPVCFromMap(map[string]string{})
+	if errArifactPVC != nil {
+		logging.FromContext(ctx).Info(errArifactPVC)
+	}
+	metrics, errMetrics := newMetricsFromMap(map[string]string{})
+	if errMetrics != nil {
+		logging.FromContext(ctx).Info(errMetrics)
+	}
+	trustedresources, errTrustedresources := NewTrustedResourcesConfigFromMap(map[string]string{})
+	if errTrustedresources != nil {
+		logging.FromContext(ctx).Info(errTrustedresources)
+	}
+	spireconfig, errSpireconfig := NewSpireConfigFromMap(map[string]string{})
+	if errSpireconfig != nil {
+		logging.FromContext(ctx).Info(errSpireconfig)
+	}
```

Comment:
thanks for doing this! I apologize but I have a few more nits here:
* instead of renaming the err variable every time its clearer to just use `err` in all cases (usually we don't want to reuse variables and change their meaning so your instinct is right but imo error variables are an exception to this - similar to what you are doing with `ok` in NewSpireConfigFromMap)
* test coverage of this file went way down after this change b/c the error conditions are not being tested
* Can you add more context to the log message, e.g. something like "Spire configuration was invalid: %v"?

Testing log message handling is super ugly and I don't recommend having tests that assert on the log content

One way that you could address most of the above would be to collect a list of errors and log once at the end - we are already using `go-multierror` in other parts of the project, e.g. in https://github.com/tektoncd/pipeline/blob/2ee148d7af881588fc50846344ff5b4d63bbf07c/pkg/pod/status.go#L155 you can see that one error object `merr` is initialized and errors are appended as the function executes. If you collect errors as you go you could log once at the end. then to test, you can have a few cases where some/all/one value is invalid, and maybe don't assert that the log function was called, but assert that the object was constructed correctly in those cases

Reply:
I think this should be made into an issue and moved to a separate PR as this is out of scope for this PR.

Reply:
+1 to @pxp928 to make this an issue.
2 reasons

1. This requires more thought. Do we need to handle the errors? The initialization is for default values so we will never get errors.
2. This is a larger fix which is unrelated to this PR.

I can add an issue to track this and other locations where errors are getting dropped.
@bobcatfish WDYT

Reply:
Sounds good to me! Looks like you already undid the update to change the error handling; my request would be that we either do nothing (which it looks like is what we're doing, and np there since it's definitely scope creep) or if we do make changes, we do so with test coverage (and addressing variable handling, etc., as discussed above)

np with leaving this out of scope and opening an issue - could be a good 'help wanted' 'good first issue' if all we want done is adding some logging